templates/ext/herb/nodes.c.erb

```diff
@@ -39,7 +39,11 @@ static VALUE rb_<%= node.human %>_from_c_struct(<%= node.struct_type %>* <%= nod
   <%- when Herb::Template::ArrayField -%>
   VALUE <%= node.human %>_<%= field.name %> = rb_nodes_array_from_c_array(<%= node.human %>-><%= field.name %>);
   <%- when Herb::Template::ElementSourceField -%>
-  VALUE <%= node.human %>_<%= field.name %> = rb_utf8_str_new_cstr(element_source_to_string(<%= node.human %>-><%= field.name %>));
+  VALUE <%= node.human %>_<%= field.name %>;
+  {
+    hb_string_T element_source_string = element_source_to_string(<%= node.human %>-><%= field.name %>);
+    <%= node.human %>_<%= field.name %> = rb_utf8_str_new(element_source_string.data, element_source_string.length);
+  }
```

Comment:
For the future, maybe we should also introduce an equivalent `create_string` function for the Ruby extension (like we did in #749).